tutorial_code/example1_2_fm_loop_correlations.py

```diff
@@ -177,77 +197,113 @@ def run_experiment(param, shim, stats, embeddings, alpha_Phi=0., alpha_J=0.):
+        shim = data_dict["shim"]
+        stats = data_dict["stats"]
     else:
-        for iter in tqdm(range(param['num_iters']), total=param['num_iters']):
-            if iter < 100:
-                shim['alpha_Phi'] = 0.
+        # prev_execution_time = 182.0092 sec.
+        print("Collection of data typically requires several minutes")
+        for iteration in tqdm(range(param["num_iters"]), total=param["num_iters"]):
+            if iteration < param["num_iters_unshimmed_flux"]:
+                shim["alpha_Phi"] = 0.0
             else:
-                shim['alpha_Phi'] = alpha_Phi
-            if iter < 200:
-                shim['alpha_J'] = 0.
+                shim["alpha_Phi"] = alpha_Phi
+            if iteration < param["num_iters_unshimmed_J"]:
+                shim["alpha_J"] = 0.0
             else:
-                shim['alpha_J'] = alpha_J
+                shim["alpha_J"] = alpha_J
             run_iteration(param, shim, stats, embeddings)
+        if use_cache:
+            save_experiment_data(prefix, {"param": param, "shim": shim, "stats": stats})
+
+    paper_plots_example1_2(
+        all_couplings=stats["all_couplings"],
+        all_fbos=stats["all_fbos"],
+        mags=stats["mags"],
+        frust=stats["frust"],
+    )
 
-        save_experiment_data(
-            prefix,
-            {'param': param, 'shim': shim, 'stats': stats}
-        )
 
-    plot_data(all_fbos=stats['all_fbos'], mags=stats['mags'],
-              all_couplings=stats['all_couplings'], frust=stats['frust'],
-              all_alpha_phi=stats['all_alpha_Phi'], all_alpha_j=stats["all_alpha_J"],
-              coupler_orbits=shim['coupler_orbits'], alpha_phi=shim['alpha_Phi'], alpha_j=shim['alpha_J'],
-              coupling=param["coupling"], L=param["L"])
-    paper_plots_example1_2(all_couplings=stats['all_couplings'], all_fbos=stats['all_fbos'])
+def main(
+    solver_name: str = None,
+    coupling: float = -0.2,
+    num_iters: int = 300,
+    num_iters_unshimmed_flux: int = 100,
+    num_iters_unshimmed_J: int = 200,
+    use_cache: bool = True,
+) -> None:
+    """Main function to run example.
 
+    Completes an experiment matched to Figure 7 of DOI:10.3389/fcomp.2023.1238988,
+    plotting a corresponding figure.
 
-def main():
-    """Main function to run example
+    Args:
+        solver_name (string): option to specify sampler type. The default client QPU
+            is used by default other options are listed in Leap, to use a locally executed
+            classical placeholder for debugging select 'MockDWaveSampler'.
+        coupling (float): coupling strength on chain.
+        num_iters (int): Number of sequential programmings.
+        num_iters_unshimmed_flux (int): Number of sequential programmings without flux shimming.
+        num_iters_unshimmed_J (int): Number of sequential programmings without J shimming.
+        max_num_emb (int): Maximum number of embeddings to use per programming.
```

Comment:
You forgot to add max_num_emb as an argument with default float('Inf')

Reply:
I thought we wanna default to 1 for this tutorial. Do we want to instead default to float('inf')?